### DISCUSSION TITLE
HTTP/2 with content longer than 16KiB

### DISCUSSION THREAD

**Discussion by janilcgarcia** [http4s/http4s#6842] [Q&A] [answered]
Hello, guys! I'm trying to integrate cats, CE, http4s and sangria to get a GraphQL server in Scala. For subscription I would rather just use fetch + HTTP/2 because I don't want to rely on WebSockets since this isn't really bi-directional communication and WebSockets is kind of obsolete anyway.

Here is the code for reference (it's a mess, sorry):
https://github.com/janilcgarcia/cats-sangria

I'm having trouble: the introspection query the client does returns normally when I'm using HTTP/1.1, but it fails by not returning the body when using HTTP/2. I experimented a little bit and found that if the response is bigger than 16KiB it doesn't return the body, but if it's less it does!

Is this misconfiguration from my part or should I report as bug?
Note: The Response entity actually contains the entire body (The logger is setup in that project so you can see the Response object in the logger), it just fails to send that back to the client.

**TL; DR: Sending responses (strict) >= 16KiB causes the client to receive an empty body, should this be reported as a bug?**

**Comment by armanbilge** [comment]
> TL; DR: Sending responses (strict) >= 16KiB causes the client to receive an empty body, should this be reported as a bug?

if you can create a minimal, reproducible example, then yes.

---

Separately, you might find Grackle interesting. It's a GraphQL server built with Cats Effect.
https://github.com/gemini-hlsw/gsp-graphql

**Reply by janilcgarcia** [reply]
So I have plans for tomorrow, then

**Comment by janilcgarcia** [comment, marked as answer]
Looks like an actual bug
